rllib/policy/sample_batch.py

```diff
@@ -196,6 +196,8 @@ def concat_samples(
                 assert s.time_major == time_major
                 if zero_padded:
                     assert s.max_seq_len == max_seq_len
+                if max_seq_len is not None:
```

Comment:
if we are checking for None, should we also check s.max_seq_len here?
I guess if any of these max_seq_len is None, the merged value should be None.
not sure where a None max_seql_len would be useful though ...

Reply:
@gjoliver Yeah that is what I was asking in the PR comments.
My thought was to do something like this:

```
if max_seq_len is not None:
    if s.max_seq_len is None:
        raise ValueError(sample_batch is missing max_seq_len")
    else:
        max_seq_len = max(max_seq_len, s.max_seq_len)

Reply:
@gjoliver Yeah that is what I was asking in the PR comments.
My thought was to do something like this:

if max_seq_len is not None:
    if s.max_seq_len is None:
        raise ValueError("sample_batch is missing max_seq_len")
    else:
        max_seq_len = max(max_seq_len, s.max_seq_len)

Reply:
@gjoliver Yeah that was what I was asking about in the PR text
```python
if max_seq_len is not None:
    if s.max_seq_len is None:
        raise ValueError("sample_batch missing max_seq_len")
else:
    max_seq_len = max(max_seq_len, s.max_seq_len)
```

I guess we would still be missing the case where the first sample_batch had max_seq_len=None but another one in the list did. Do you want to check for that too?

Reply:
I guess we can either not check for None at all, and see where and why things may break, so simply:

max_seq_len = max(max_seq_len, s.max_seq_len)

or check for None on both sides, like:

if max_seq_len is None or s.max_seq_len is None:
    max_seq_len = None
else:
    max_seq_len = max(…, …)

what do you think?

Reply:
I decided to handle both caseI updated it to look like this. What do you think?
```python
if (s.max_seq_len is None or max_seq_len is None)\
                   and s.max_seq_len != max_seq_len:
                    raise ValueError(
                        "Samples must consistently provide or omit max_seq_len"
                    )
                if zero_padded:
                    assert s.max_seq_len == max_seq_len
                if max_seq_len is not None:
                    max_seq_len = max(max_seq_len, s.max_seq_len)
                concat_samples.append(s)
```

Reply:
seems complete. hope we uncover more bugs :)
thanks!!